### PR TITLE
Document weave/bns tx format

### PR DESCRIPTION
**Pull request by orkunkl**
This pr closes issue iov-one/weave#840

### CODE REVIEW

File: docs/transaction.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+## Protocol buffers
+---
+Let's start with explaining what is protobuf and how to use them:
+
+> Protocol Buffers (a.k.a., protobuf) are Google's language-neutral, platform-neutral, extensible mechanism for serializing structured data.
+
+This line is from [protobuf github repo](https://github.com/protocolbuffers/protobuf)
+
+In a nutshell, protobuf is a effective and easy to use communication protocol. To use it define protoc definitions(*.proto*) and then compile them to a desired language which protocol buffers [supports](https://developers.google.com/protocol-buffers/docs/tutorials). After these steps, high tech comms is ready to use the in your software.
```
Comment (ethanfrey) [inline docs/transaction.md]: s/communication protocol/serialization format/

File: docs/transaction.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+```
+
+`oneof sum` in the _tx message_ means, one of the messages will be sent to *bnsd*. *reference:* [developers.google.com/protocol-buffers](https://developers.google.com/protocol-buffers/docs/proto3#oneof)
+
+## Communicating with Weave/bnsd
+---
+To have required tools to communicate and send transactions to _weave/bnsd_, follow these steps
+
+### 1 - Import _bnsd_ protobuf definitios  
+- Download [spec/proto/cmd/bnsd/app/codec.proto](https://github.com/iov-one/weave/blob/v0.18.0/spec/proto/cmd/bnsd/app/codec.proto) under _spec_ folder on project's root.
```
Comment (ethanfrey) [inline docs/transaction.md]: You need to copy the whole *spec/proto* directory into the repo, as the imports should refer to local files

File: docs/transaction.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+### 2 - Install your preferred language-specific protoc plugin
+> We recommend using [uber/prototool](https://github.com/uber/prototool) or better [IOV/prototool-docker](https://github.com/iov-one/prototool-docker). Import _prototool-docker_ directly from [docker hub](https://hub.docker.com/r/iov1/prototool)
```
Comment (ethanfrey) [inline docs/transaction.md]: :+1:

**Comment by ethanfrey** [comment]
Oh, this should probably be in a subdirectory (weave api spec?)
And that subdir linked from the homepage in an obvious way.

Then we can provide link to users

File: docs/transaction.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
+# Weave transactions
+
+_Weave/BNS_ uses protobuf for handling communications
```
Comment (husio) [inline docs/transaction.md]: Period at the end of the sentence is missing.

File: docs/transaction.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+
+For the beginning, check out the *proto* files and have an idea what are they and how they look like.
+
+## Protocol buffers
+---
```
Comment (husio) [inline docs/transaction.md]: I think section (`##`) is enough for separation, horizontal line `---` is not needed.

File: docs/transaction.md (hunk 1 above)
Comment (husio) [inline docs/transaction.md]: is **an** effective

**Pull request merged**